### ISSUE TITLE
player-accessed-on-wrong-thread

### ISSUE THREAD

**Issue by sandeep14** [TheWidlarzGroup/react-native-video#2966]
# Bug

below is error reported on playstore console
Exception java.lang.IllegalStateException: Player is accessed on the wrong thread. See https://exoplayer.dev/issues/player-accessed-on-wrong-thread
  at com.google.android.exoplayer2.SimpleExoPlayer.verifyApplicationThread (SimpleExoPlayer.java)
  at com.google.android.exoplayer2.SimpleExoPlayer.getPlayWhenReady (SimpleExoPlayer.java)
  at com.brentvatne.exoplayer.ReactExoplayerView.pausePlayback (ReactExoplayerView.java)
  at com.brentvatne.exoplayer.ReactExoplayerView.onAudioFocusChange (ReactExoplayerView.java)
  at android.media.AudioManager$ServiceEventHandlerDelegate$1.handleMessage (AudioManager.java:3017)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:246)
  at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run (MessageQueueThreadImpl.java)
  at java.lang.Thread.run (Thread.java:923)

## Platform
android 

Which player are you experiencing the problem on:

* Android

## Environment info
react-version 0.70.6

Library version: 5.2.1

**Comment by freeboub** [comment]
can it be reproduced on master branch ?

**Comment by kyle-ssg** [comment]
I can confirm this is replicable 6.0.0-alpha.6 at least, though in rare circumstances. I have only 1 Sentry report of this 

Redmi Note 10 Pro
Android 11

**Comment by freeboub** [comment]
@kyle-ssg can you provide the callstack please?

**Comment by kyle-ssg** [comment]
![image](https://github.com/react-native-video/react-native-video/assets/8608314/5cdd22f2-457a-439d-9086-824703fbb4fe)

 Hope this helps, thanks so much for all the effort into this library.

**Comment by freeboub** [comment]
This issue will be fixed in 6.0.0-alpha.9

Let me know if you have other crash reposted by crashlytics, it is always interesting !

**Comment by PadovaY** [comment]
hey, this crash happens to me in release mode, mostly on android 13/14 devices. version: 6.2.0.
any suggestion?

stacktrace:
```
Fatal Exception: java.lang.IllegalStateException: Player is accessed on the wrong thread.
Current thread: 'mqt_native_modules'
Expected thread: 'main'
See https://developer.android.com/guide/topics/media/issues/player-accessed-on-wrong-thread
       at androidx.media3.exoplayer.ExoPlayerImpl.verifyApplicationThread(ExoPlayerImpl.java:2798)
       at androidx.media3.exoplayer.ExoPlayerImpl.getPlayWhenReady(ExoPlayerImpl.java:817)
       at com.brentvatne.exoplayer.ReactExoplayerView.pausePlayback(ReactExoplayerView.java:1257)
       at com.brentvatne.exoplayer.ReactExoplayerView.-$$Nest$mpausePlayback()
       at com.brentvatne.exoplayer.ReactExoplayerView$OnAudioFocusChangedListener.onAudioFocusChange(ReactExoplayerView.java:1180)
       at android.media.AudioManager$ServiceEventHandlerDelegate$1.handleMessage(AudioManager.java:4318)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loopOnce(Looper.java:230)
       at android.os.Looper.loop(Looper.java:319)
       at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:233)
       at java.lang.Thread.run(Thread.java:1012)
```

**Comment by freeboub** [comment]
@PadovaY here is a fix for the issue: https://github.com/TheWidlarzGroup/react-native-video/pull/3916

**Comment by forchello** [comment]
got this crash in:
- android 12
- react native 0.73.8
- react native video 6.1.2

 FATAL EXCEPTION: mqt_native_modules
                                                                                                    Process: com.passionclub.love.stories, PID: 18495
                                                                                                    java.lang.IllegalStateException: Player is accessed on the wrong thread.
                                                                                                    Current thread: 'mqt_native_modules'
                                                                                                    Expected thread: 'main'
                                                                                                    See https://developer.android.com/guide/topics/media/issues/player-accessed-on-wrong-thread
                                                                                                    	at androidx.media3.exoplayer.ExoPlayerImpl.verifyApplicationThread(ExoPlayerImpl.java:2798)
                                                                                                    	at androidx.media3.exoplayer.ExoPlayerImpl.getPlayWhenReady(ExoPlayerImpl.java:817)
                                                                                                    	at com.brentvatne.exoplayer.ReactExoplayerView.pausePlayback(ReactExoplayerView.java:1211)
                                                                                                    	at com.brentvatne.exoplayer.ReactExoplayerView.-$$Nest$mpausePlayback(Unknown Source:0)
                                                                                                    	at com.brentvatne.exoplayer.ReactExoplayerView$OnAudioFocusChangedListener.onAudioFocusChange(ReactExoplayerView.java:1137)
                                                                                                    	at android.media.AudioManager$ServiceEventHandlerDelegate$1.handleMessage(AudioManager.java:2962)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                                                    	at android.os.Looper.loop(Looper.java:236)
                                                                                                    	at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:234)

**Comment by forchello** [comment]
> @PadovaY here is a fix for the issue: #3916

Thanks, works with 6.4.3. But now I have black screen while loading the video. backgroundColor: 'transparent' does not help :)

**Comment by forchello** [comment]
> @PadovaY here is a fix for the issue: #3916

The latest version has stopped crashing, but I have a black screen flashing in android. And yes, I changed the shutterColor to transparent, but nothing happened.
